content/blog/2020/07/2020-07-30-github-checks-api-plugin-coding-phase-2.adoc

```diff
@@ -0,0 +1,95 @@
+Without leaving GitHub, users are now able to see the analysis report they interested in.
+
+image::/images/post-images/2020/07-github-checks-api-plugin-coding-phase-2/warning-checks.png[Warning Checks Summary]
+
+On GitHub's conversation tab for each PR, users will see summaries for those checks like the screenshot above. The summaries will include:
+
+* The status that indicates the quality gate
+* The name of the analysis tool used
+* A short message that indicates statistics of new and total issues
+
+More fine-grained statistics can be found in the `Details` page.
+
+image::/images/post-images/2020/07-github-checks-api-plugin-coding-phase-2/severity-statistics.png[Severity Statis]
+
+Another practical feature is the annotation for specific lines of code. Users can now review the code alone with the annotations.
+
+image::/images/post-images/2020/07-github-checks-api-plugin-coding-phase-2/annotations.png[Warning Annotations]
+
+=== Try It
+
+In https://github.com/jenkinsci/warnings-ng-plugin/releases/tag/warnings-ng-8.4.0[Wanings NG plugin 8.4.0], the warning checks is set as a default feature. 
+Therefore, you can get those checks for your own GitHub project just in a few steps:
+
+1. Update Warnings NG plugin to 8.4.0
+2. Install link:https://plugins.jenkins.io/checks-api/[Checks API plugin] and link:https://plugins.jenkins.io/github-checks/[GitHub Checks plugin] on your Jenkins instance
+3. Configure warnings-ng plugin as usual for a multi-branch project or a GitHub organization project
+
+=== Pipeline Usage:
+
+Pipeline users can control this feature through a boolean field `skipPublishingChecks`, e.g.
+
+```
+recordIssues tool: errorProne(), healthy: 1, unhealthy: 20, skipPublishingChecks: false
+```
+
+== Coverage Checks
+
+The coverage checks are achieved by consuming the API in link:https://plugins.jenkins.io/code-coverage-api/[Code Coverage API plugin].
+First, in the conversation tab of a PR, users will be able to see the summary about the coverage and the comparing with previous builds.
+
+image::/images/post-images/2020/07-github-checks-api-plugin-coding-phase-2/coverage-summary.png[Coverage Summary]
+
+Besides, the `Details` page will contain some other things:
+
+* The links to the reference build, including the target branch build from the master branch and the last successful build from this branch
+* Coverage healthy score (the default value is 100% if the threshold is not configured)
+* Coverages and trends of different types in table format
+
+image::/images/post-images/2020/07-github-checks-api-plugin-coding-phase-2/coverage-details.png[Coverage Details]
+
+The link:https://github.com/jenkinsci/code-coverage-api-plugin/pull/169[pull request] for this feature will soon be merged and will be included in the next release of Coverage Checks API plugin.
+
+== Next Phase
```

Comment:
Would it make sense to mention that we are also working on making the feature available on ci.jenkins.io for plugin maintainers? https://issues.jenkins-ci.org/browse/INFRA-2694

Once completed, it would be a major feature to showcase IMHO. CC @timja @uhafner